docs/design.md

```diff
@@ -4,42 +4,65 @@ by Spencer Kimball from early 2014.
-cockroach servers. Ranges are defined by start and end keys. Ranges are
-merged and split to maintain total byte size within a globally
-configurable min/max size interval. Range sizes default to target `64M` in
-order to facilitate quick splits and merges and to distribute load at
-hotspots within a key range. Range replicas are intended to be located
-in disparate datacenters for survivability (e.g. `{ US-East, US-West,
-Japan }`, `{ Ireland, US-East, US-West}`, `{ Ireland, US-East, US-West,
-Japan, Australia }`).
-
-Single mutations to ranges are mediated via an instance of a distributed
-consensus algorithm to ensure consistency. We’ve chosen to use the
-[Raft consensus algorithm](https://raftconsensus.github.io); all consensus
-state is stored in RocksDB.
-
-A single logical mutation may affect multiple key/value pairs. Logical
-mutations have ACID transactional semantics. If all keys affected by a
-logical mutation fall within the same range, atomicity and consistency
-are guaranteed by Raft; this is the **fast commit path**. Otherwise, a
-**non-locking distributed commit** protocol is employed between affected
-ranges.
+CockroachDB is a distributed SQL database. The primary design goals
+are **scalability**, **strong consistency** and **survivability**
+(hence the name). Cockroach aims to tolerate disk, machine, rack, and
+even **datacenter failures** with minimal latency disruption and **no
+manual intervention**. Cockroach nodes are symmetric; a design goal is
+**homogeneous deployment** (one binary) with minimal configuration and
+no required external dependencies.
+
+The entry point for database clients is the SQL interface. Every node
+in a CockroachDB cluster can act as a client SQL gateway. A SQL
+gateway transforms and executes client SQL statements to key-value
+(KV) operations, which the gateway distributes across the cluster as
+necessary and returns results to the client. CockroachDB implements a
+**single, monolithic sorted map** from key to value where both keys
+and values are byte strings.
+
+The KV map is logically composed of smaller segments of the keyspace
+called ranges. Each range is backed by data stored in a local KV
+storage engine (we use [RocksDB](http://rocksdb.org/), a variant of
+LevelDB). Range data is replicated to a configurable number of
+additional CockroachDB nodes. Ranges are merged and split to maintain
+a target size, by default `64M`. The relatively small size facilitates
+quick repair and rebalancing to address node failures, new capacity
+and even read/write load.
+
+CockroachDB achieves horizontally scalability:
+- adding more nodes increases the capacity of the cluster by the
+  amount of storage on each node (divided by a configurable
+  replication factor), theoretically up to 4 exabytes (4E) of logical
+  data;
+- client queries can be sent to any node in the cluster, and queries
+  can operate fully independently from each other, meaning that
```

Comment:
The queries can operate independently when there aren't conflicts. Not sure if that detail is worth calling out here.